### PR TITLE
Bump dependencies, reduce plugin size

### PR DESCRIPTION
**Pull request by detouched**
- Bump dependencies to their recent releases
- Remove unused dependencies and transitive dependencies
- Avoid packing dependencies provided by IntelliJ platform
- Updgrade ktlint, move indent configuration to `.editorconfig`
- Format the code with new ktlint

### CODE REVIEW

**Comment by detouched** [comment]
I was going to fix a couple of things in the plugin (UI freeze, project-level config) but decided to refresh it a little bit first.

Removing unused libraries also helped to shrink the plugin zip from 17Mb to 10Mb.

File: .editorconfig [hunk 1]
```diff
@@ -1,3 +1,4 @@
 [{*.kt,*.kts}]
-disabled_rules = no-wildcard-imports
-ij_kotlin_imports_layout = *
\ No newline at end of file
+indent_size = 4
```
Comment (detouched) [inline .editorconfig:2]: Recent version of [kotlinter](https://github.com/jeremymailen/kotlinter-gradle) doesn't have indent setting anymore, hence I moved it from `build.gradle.kts` to `.editorconfig`.

File: .editorconfig [hunk 2]
```diff
@@ -1,3 +1,4 @@
...
+ktlint_disabled_rules = no-wildcard-imports
```
Comment (detouched) [inline .editorconfig:3]: This property has been renamed, ktlint warned about it.

File: build.gradle.kts [hunk 3]
```diff
@@ -29,18 +29,25 @@ repositories {
...
 
+configurations.all {
+    // Dependencies provided by IntelliJ runtime
+    exclude("com.google.code.gson", "gson")
+    exclude("org.slf4j", "slf4j-api")
+}
+
 dependencies {
-    implementation("io.pebbletemplates:pebble:3.1.5")
-    implementation("org.junit.jupiter:junit-jupiter:5.8.1")
```
Comment (detouched) [inline build.gradle.kts:34]: This should be a dependency only for tests, not needed in the plugin ZIP.

File: build.gradle.kts [hunk 4]
```diff
@@ -29,18 +29,25 @@ repositories {
...
+        exclude("org.jetbrains.kotlinx", "kotlinx-coroutines-core")
+    }
+    implementation("com.google.code.gson:gson:2.10")
     implementation("com.launchdarkly:api-client:3.10.0")
     implementation("com.launchdarkly:launchdarkly-java-server-sdk:5.+")
-    implementation("com.google.code.gson:gson:2.7")
-    implementation("com.googlecode.json-simple", "json-simple", "1.1.1")
-    implementation("com.github.doyaaaaaken:kotlin-csv-jvm:0.14.0")
-    runtimeOnly("org.jetbrains.kotlin:kotlin-reflect:1.4.20-M2")
-    implementation("org.apache.logging.log4j", "log4j-slf4j-impl", "2.17.1")
```
Comment (detouched) [inline build.gradle.kts:42]: Not sure why `kotlin-reflect` or log4j were explicitly included. They are not pulled in by any other dependencies, and the plugin seems to be working fine without them.

Removing these saved almost 5Mb.

File: build.gradle.kts [hunk 5]
```diff
@@ -29,18 +29,25 @@ repositories {
...
-    detektPlugins("io.gitlab.arturbosch.detekt:detekt-formatting:1.10.0-RC1")
+    implementation("com.github.doyaaaaaken:kotlin-csv-jvm:1.6.0") {
+        // Not using any of the async functionality
+        exclude("org.jetbrains.kotlinx", "kotlinx-coroutines-core")
+    }
+    implementation("com.google.code.gson:gson:2.10")
     implementation("com.launchdarkly:api-client:3.10.0")
     implementation("com.launchdarkly:launchdarkly-java-server-sdk:5.+")
-    implementation("com.google.code.gson:gson:2.7")
-    implementation("com.googlecode.json-simple", "json-simple", "1.1.1")
```
Comment (detouched) [inline build.gradle.kts:39]: This one was also unused.

File: build.gradle.kts [hunk 6]
```diff
@@ -29,18 +29,25 @@ repositories {
...
+    exclude("org.slf4j", "slf4j-api")
+}
+
 dependencies {
-    implementation("io.pebbletemplates:pebble:3.1.5")
-    implementation("org.junit.jupiter:junit-jupiter:5.8.1")
-    detektPlugins("io.gitlab.arturbosch.detekt:detekt-formatting:1.10.0-RC1")
+    implementation("com.github.doyaaaaaken:kotlin-csv-jvm:1.6.0") {
+        // Not using any of the async functionality
+        exclude("org.jetbrains.kotlinx", "kotlinx-coroutines-core")
```
Comment (detouched) [inline build.gradle.kts:41]: The CSV reader used by LD plugin doesn't touch any of the asynchronous functionality of the library, so I believe coroutines can be excluded. Unfortunately I can't verify this with Code References feature, but I tried reading some irrelevant CSV file using the same method elsewhere and that worked just fine.

This saves around 1.4Mb.

File: src/main/kotlin/com/launchdarkly/intellij/hover/HoverDocumentationProvider.kt [hunk 7]
```diff
@@ -20,7 +20,7 @@ import java.io.StringWriter
 
 class HoverDocumentationProvider : AbstractDocumentationProvider() {
     private fun getFlagConfig(contextElement: PsiElement, flagKey: String): FlagConfiguration? {
-        return contextElement?.project?.service<FlagStore>()?.flagConfigs[flagKey]
+        return contextElement.project.service<FlagStore>().flagConfigs[flagKey]
```
Comment (detouched) [inline src/main/kotlin/com/launchdarkly/intellij/hover/HoverDocumentationProvider.kt:23]: Using bracket notation on nullable object became an error in recent Kotlin. Could've replaced with `.get(flagKey)`, but `contextElement` is guaranteed to be non-null, so entire null-safe chain in fact had no reason.

File: build.gradle.kts [hunk 8]
```diff
@@ -57,11 +64,10 @@ intellij {
     pluginName.set(properties("pluginName"))
     version.set(properties("platformVersion"))
     type.set(properties("platformType"))
-    //localPath.set("/Applications/IntelliJ IDEA.app")
+    // localPath.set("/Applications/IntelliJ IDEA.app")
```
Comment (detouched) [inline build.gradle.kts:67]: This and all changes below (except one with null-safe chain) are the outcome of linting the code with recent ktlint.

**Review by monsagri — COMMENTED** [review]
Hi @detouched!
Thanks for the PR and also the general interest! 
Just going to go through this now and look to approve/properly review asap!

**Review by monsagri — APPROVED** [review]
LGTM! 
Thanks for a very clear and readable PR!

**Pull request merged**